### PR TITLE
Add sync

### PR DESCRIPTION
**Pull request by Nepomuceno**
### What does this PR do?

Adding sync endpoint to the backend to import metadata files automatically.

### All Submissions:

* [x] Have you followed the guidelines in our Contributing document?
* [x] Have you successfully ran tests with your changes locally? Including container validation.
* [x] Have you lint your code locally prior to submission?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../../pulls) for the same update/change?
* [x] Have you labelled this PR correctly? E.G. major, milestone, breaking, breaking-change, minor, feature, enhancement, patch, fix, chore, refactor etc

### CODE REVIEW

File: api/blob/azure_client.py [hunk 1]
```diff
@@ -82,3 +90,20 @@ async def get_new_thumbnail(self, data_type: str, filepath: str) -> bytes:
         content = await self.get_blob_content(iq_path, 8000, fftSize * 512)
         image = get_spectrogram_image(content, data_type, fftSize)
         return image
+
+    async def get_medatada_files(self):
```
Comment (hannahlspencer) [inline api/blob/azure_client.py:94]: typo in metadata

File: api/blob/azure_client.py [hunk 2]
```diff
@@ -82,3 +90,20 @@ async def get_new_thumbnail(self, data_type: str, filepath: str) -> bytes:
...
+        # files that enf with .sigmf-meta
+        async for blob in container_client.list_blobs():
+            if blob.name.endswith(".sigmf-meta"):
+                blob_client = self.get_blob_client(blob.name)
+                blob = await blob_client.download_blob()
+                content = await blob.readall()
+                yield str(blob.name), Metadata.parse_raw(content)
+        return
+
+    async def get_metadata_file(self, filepath: str):
```
Comment (hannahlspencer) [inline api/blob/azure_client.py:105]: is there a plan for when this would be used?

File: api/handlers/datasources.py [hunk 3]
```diff
@@ -125,3 +125,22 @@ async def update_datasource(
...
+
+
+@router.put("/api/datasources/{account}/{container}/sync", status_code=204)
+async def sync_datasource(
+    account: str,
+    container: str,
+    datasources_collection: AgnosticCollection = Depends(datasource_repo.collection),
+    backgound_tasks=BackgroundTasks,
+):
+    existingDatasource = await datasources_collection.find_one(
```
Comment (Jade-Codes) [inline api/handlers/datasources.py:137]: snake case

File: api/blob/azure_client.py (hunk 2 above)
Comment (Nepomuceno) [reply api/blob/azure_client.py:105]: This is used already by the sync method that was introduced as part of this pr.

Comment (Nepomuceno) [reply api/blob/azure_client.py:105]: Sorry misread the name you were talking about getting a single file.yeah I was using ti before but not anymore but would make sense so strandarize and make it onwe method will group thodse thx

Comment (Nepomuceno) [reply api/blob/azure_client.py:103]: Fixed

File: api/handlers/datasources.py (hunk 3 above)
Comment (Nepomuceno) [reply api/handlers/datasources.py:139]: Fixed

**Review by Jade-Codes — APPROVED** [review]

**Pull request merged**